### PR TITLE
Navigation for receive revamp

### PR DESCRIPTION
**Pull request by stackchain**

### CODE REVIEW

File: src/TxHistory/WalletHero/ActionsBanner.tsx [hunk 1]
```diff
@@ -121,7 +121,7 @@ const useNavigations = () => {
 
   return {
     onSend: () => navigation.navigate(WALLET_ROUTES.SEND),
-    onReceive: () => navigation.navigate(WALLET_ROUTES.RECEIVE),
+    onReceive: () => navigation.navigate('ReceiveScreen'),
```
Comment (stackchain) [inline src/TxHistory/WalletHero/ActionsBanner.tsx]: keep our standard

File: src/Receive/Context/ReceiveContext.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+export const useReceiveContext = () => React.useContext(ReceiveContext) || missingProvider()
+
+export const useReceiveContextInfoModal = () => {
+  const [receiveContextState, setReceiveContextState] = useReceiveContext()
+  const showInfoModal = () => setReceiveContextState({isInfoModalOpen: true})
+  const hideInfoModal = () => setReceiveContextState({isInfoModalOpen: false})
+
+  return {isInfoModalOpen: receiveContextState?.isInfoModalOpen, showInfoModal, hideInfoModal}
```
Comment (stackchain) [inline src/Receive/Context/ReceiveContext.tsx]: get rid of question mark

File: src/Receive/Context/ReceiveContext.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+const missingProvider = () => {
+  throw new Error('ReceiveProvider is missing')
+}
+
+export const useReceiveContext = () => React.useContext(ReceiveContext) || missingProvider()
+
+export const useReceiveContextInfoModal = () => {
+  const [receiveContextState, setReceiveContextState] = useReceiveContext()
+  const showInfoModal = () => setReceiveContextState({isInfoModalOpen: true})
```
Comment (stackchain) [inline src/Receive/Context/ReceiveContext.tsx]: wrap it in useCallback

File: src/Receive/Context/ReceiveContext.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import * as React from 'react'
+
+type ReceiveContextState = {
```
Comment (stackchain) [inline src/Receive/Context/ReceiveContext.tsx]: simplify to boolean

File: src/navigation.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  headerStyle: {
+    elevation: 0,
+    shadowOpacity: 0,
+    backgroundColor,
+  },
+  title,
+  headerRight,
+})
+
+export const SettingsButton = () => {
```
Comment (stackchain) [inline src/navigation.tsx]: Buttons Gallery

File: src/Receive/ReceiveScreenNavigator.tsx [hunk 6]
```diff
@@ -17,31 +18,33 @@ export const ReceiveScreenNavigator = () => {
...
-          headerRight: () => (
-            <Button
-              style={styles.settingsButton}
-              onPress={() => navigation.navigate(WALLET_ROOT_ROUTES.SETTINGS)}
-              iconImage={iconGear}
-              title=""
-              withoutBackground
-            />
-          ),
+    <ReceiveProvider>
```
Comment (stackchain) [inline src/Receive/ReceiveScreenNavigator.tsx]: Check if is necessary

File: src/TxHistory/TxHistoryNavigator.tsx [hunk 7]
```diff
@@ -12,79 +12,91 @@ import {transactionsInfoSelector, walletMetaSelector} from '../../legacy/selecto
...
 
 export const TxHistoryNavigator = () => {
+  const strings = useStrings()
   const walletMeta = useSelector(walletMetaSelector)
   const transactionInfos = useSelector(transactionsInfoSelector)
-
   return (
-    <Stack.Navigator screenOptions={defaultStackNavigatorOptions} initialRouteName={TX_HISTORY_ROUTES.MAIN}>
-      <Stack.Screen name={TX_HISTORY_ROUTES.MAIN} component={TxHistory} options={headerBarOptions(walletMeta.name)} />
+    <ReceiveProvider>
```
Comment (stackchain) [inline src/TxHistory/TxHistoryNavigator.tsx]: Lift it up for now

File: src/TxHistory/TxHistoryNavigator.tsx [hunk 8]
```diff
@@ -26,20 +27,20 @@ export const TxHistoryNavigator = () => {
...
+    <ModalInfoProvider>
       <Stack.Navigator screenOptions={defaultStackNavigatorOptions} initialRouteName={TX_HISTORY_ROUTES.MAIN}>
         <Stack.Screen
-          name="TxHistory"
+          name="history-list"
           component={TxHistory}
           options={
             UI_V2
-              ? buildOptionsWithDefault({title: walletMeta.name, headerRight: SettingsButton})
+              ? buildOptionsWithDefault({title: walletMeta.name, headerRight: HeaderRightHistory})
```
Comment (stackchain) [inline src/TxHistory/TxHistoryNavigator.tsx]: wrap function

File: src/TxHistory/TxHistoryNavigator.tsx [hunk 9]
```diff
@@ -48,16 +49,17 @@ export const TxHistoryNavigator = () => {
         />
 
         <Stack.Screen
-          name="ReceiveScreen"
+          name="receive"
           component={ReceiveScreen}
           options={buildOptionsWithDefault({
             title: strings.receiveTitle,
-            headerRight: ShowReceiveInfoButton,
+            headerRight: HeaderRightReceive,
```
Comment (stackchain) [inline src/TxHistory/TxHistoryNavigator.tsx]: same

**Review by wolverineks — APPROVED** [review]

**Pull request merged**